var/www/so/capme/.inc/callback-elastic.php

```diff
@@ -317,6 +317,15 @@ function elastic_command(){
 					$type = "bro_conn";
 					$bro_query = $elastic_response_object["hits"]["hits"][0]["_source"]["uid"];
 					elastic_command();
+					
+					// Check for common error conditions.
+					if (json_last_error() !== JSON_ERROR_NONE) {
+						$errMsgElastic = "Couldn't decode JSON from second ES query.";
+					} elseif ( ! isset($elastic_response_object["hits"]["total"]) ) {
+						$errMsgElastic = "Second ES query didn't return a total number of hits.";
+					} elseif ( $elastic_response_object["hits"]["total"] == "0") {
+						$errMsgElastic = "Second ES query couldn't find this ID.";
+					}
```

Comment:
If we hit one of these error conditions, then we should not continue on to line 331.  That's the reason for the if-elseif-elseif-else logic here:
https://github.com/weslambert/elastic-test/blob/9680fb5ce701f2d0e06b2cfe92bb8f5790dc6c42/var/www/so/capme/.inc/callback-elastic.php#L308-L314

So how about this?

What if we take out this:
https://github.com/weslambert/elastic-test/blob/9680fb5ce701f2d0e06b2cfe92bb8f5790dc6c42/var/www/so/capme/.inc/callback-elastic.php#L315-L329

and transplant the logic up to line 302:
https://github.com/weslambert/elastic-test/blob/9680fb5ce701f2d0e06b2cfe92bb8f5790dc6c42/var/www/so/capme/.inc/callback-elastic.php#L302

So maybe something like this?
```
                // If bro_files, we need to query Elastic and get the log
		if ($errMsgElastic == "" && $type == "bro_files") {
			elastic_command();
			
			// Check for common error conditions.
			if (json_last_error() !== JSON_ERROR_NONE) { 
				$errMsgElastic = "Couldn't decode JSON from second ES query.";
			} elseif ( ! isset($elastic_response_object["hits"]["total"]) ) {
				$errMsgElastic = "Second ES query didn't return a total number of hits.";
			} elseif ( $elastic_response_object["hits"]["total"] == "0") {
				$errMsgElastic = "Second ES query couldn't find this ID.";
                        } else {
                                // If we received a bro_files record back, we need to grab the CID and get ready to query ES again
				if ( $elastic_response_object["hits"]["hits"][0]["_source"]["type"] == "bro_files" ) {
					$type = "bro_conn";
					$bro_query = $elastic_response_object["hits"]["hits"][0]["_source"]["uid"];
                        }
                  }
```

Reply:
Yeah, I didn't even realize that it was still continuing on after the check.  That looks good -- apologies for the ignorance.  I'll adjust and push another commit.